### ISSUE TITLE
Login fails with "No such file or directory (os error 2)" on WSL2

### ISSUE THREAD

**Issue by robo9k** [nickelc/modiom#1]
Hej,

I've tried to install `modiom` as per the `README` instructions with
```
$ git clone https://github.com/nickelc/modiom.git
$ cargo install --path modiom
```

First oddity is lack of a version number:
```
$ modiom --version
modiom
```
So for reference I'm at commit ed91ce5d07b0e3107e39d5a0df7d70914a6a51e0

I'm running this on Ubuntu 20.04.1 LTS in WSL2. If you need more detail, please tell me what.

On to the actual problem, login doesn't work and consistently fails with the same error:
```
$ modiom login
Please visit https://mod.io/apikey and paste the API key below
Enter api key: ***
Enter email: ***
Authentication code request was successful.
Enter security code: ***
No such file or directory (os error 2)
```
Running with `strace` yields the following:
```
mkdir("/home/robo9k/.modio", 0777)         = -1 EEXIST (File exists)
statx(0, NULL, AT_STATX_SYNC_AS_STAT, STATX_ALL, NULL) = -1 EFAULT (Bad address)
statx(AT_FDCWD, "/home/robo9k/.modio", AT_STATX_SYNC_AS_STAT, STATX_ALL, {stx_mask=STATX_ALL, stx_attributes=0, stx_mode=S_IFDIR|0755, stx_size=4096, ...}) = 0
openat(AT_FDCWD, "/home/robo9k/.modio/credentials", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
mkdir("/home/robo9k/.modio", 0777)         = -1 EEXIST (File exists)
statx(AT_FDCWD, "/home/robo9k/.modio", AT_STATX_SYNC_AS_STAT, STATX_ALL, {stx_mask=STATX_ALL, stx_attributes=0, stx_mode=S_IFDIR|0755, stx_size=4096, ...}) = 0
openat(AT_FDCWD, "/home/robo9k/.modio/credentials", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
write(2, "No such file or directory", 25No such file or directory) = 25
write(2, " (os error ", 11 (os error )             = 11
```

Not sure what's going on there:
```
$ ls -lah ~/.modio/
total 8.0K
drwxr-xr-x  2 robo9k robo9k 4.0K Nov  1 19:29 .
drwxr-xr-x 18 robo9k robo9k 4.0K Nov  1 19:29 ..
```

Not sure if this is relevant, but here's `cargo` and `rustc` version:
```
$ cargo --version
cargo 1.47.0 (f3c7e066a 2020-08-28)
$ rustc --version
rustc 1.47.0 (18bf6b4f0 2020-10-07)
```

Let me know if I can debug this on my end.
In the meantime I'd also be happy to know what to put in `~/.modio/credentials` so I can try to use the other features :)

**Comment by nickelc** [comment]
Thanks for the bug report. The login should work if you create an empty `credentials` file.

This is how the file should look like after the login was successful for the test & prod env.
```toml
[host."https://api.mod.io/v1"]
api_key = "***********"
token = "************************************"

[host."https://api.test.mod.io/v1"]
api_key = "***********"
token = "************************************"
```

**Comment by robo9k** [comment]
Hi!

I can confirm that creating an empty `credentials` file fixed the login command issue.
```
$ touch ~/.modio/credentials
```
Its contents also look like you describe after I successfully logged in.

Is this a general workflow issue or just when creating the file?
Can I help you debugging this?

**Comment by nickelc** [comment]
It's ok. the error is easy to spot. I'm working on a fix.

**Comment by nickelc** [comment]
It should be fixed with 8a2db4f

**Comment by robo9k** [comment]
I can confirm that login works fine in latest `HEAD`:
```
$ git clone https://github.com/nickelc/modiom.git
$ cd modiom/
$ git rev-parse HEAD
008dac3cfb0b197692e7c0c68ebe59b1b029eb17
$ cargo install --path .
$ modiom --version
modiom
$ modiom login
Please visit https://mod.io/apikey and paste the API key below
Enter api key: ***
Enter email: ***
Authentication code request was successful.
Enter security code: ***
$
```

If there's nothing else that you need to do, feel free to close the issue as solved. Thanks! :)
